src/write.jl

```diff
@@ -1,140 +1,119 @@
-write_bytes(io::IO, x::AbstractFloat) = Base.write(io, string(isinteger(x) ? trunc(Int, x) : x))
-write_bytes(io::IO, continuous::Bool) = Base.write(io, continuous ? "EDF+C" : "EDF+D")
-write_bytes(::IO, ::Nothing) = 0
+_edf_write(io::IO, value::Union{String,Char}) = Base.write(io, value)
+_edf_write(io::IO, date::Date) = Base.write(io, uppercase(Dates.format(date, dateformat"dd-u-yyyy")))
+_edf_write(io::IO, date::DateTime) = Base.write(io, Dates.format(date, dateformat"dd\.mm\.yyHH\.MM\.SS"))
+_edf_write(io::IO, x::Integer) = Base.write(io, string(x))
+_edf_write(io::IO, x::AbstractFloat) = Base.write(io, string(isinteger(x) ? trunc(Int, x) : x))
 
-function write_bytes(io::IO, metadata::T) where T<:Union{PatientID,RecordingID}
+function _edf_write(io::IO, metadata::T) where T<:Union{PatientID,RecordingID}
     bytes_written = 0
     for name in fieldnames(T)
         field = getfield(metadata, name)
         if T <: RecordingID && name === :startdate
-            bytes_written += write_bytes(io, "Startdate ")
+            bytes_written += _edf_write(io, "Startdate ")
         end
-        bytes_written += write_bytes(io, field) + Base.write(io, 0x20)
-    end
-    return bytes_written
-end
-
-function write_bytes(io::IO, annotations::T) where {T<:Union{DataRecord, Vector{<:AbstractAnnotation}}}
-    bytes_written = 0
-    for annotation in annotations
-        bytes_written += write_bytes(io, annotation)
+        bytes_written += field isa Missing ? Base.write(io, 'X') : _edf_write(io, field)
+        bytes_written += Base.write(io, ' ')
     end
     return bytes_written
 end
 
-function write_bytes(io::IO, annotation::AbstractAnnotation)
-    bytes_written = Base.write(io, sign(annotation)) +
-                    write_bytes(io, annotation.offset)
-    if has_duration(annotation)
-        bytes_written += Base.write(io, 0x15) +
-                         write_bytes(io, duration(annotation))
-    end
-    bytes_written += Base.write(io, footer(annotation)...)
-    if has_events(annotation)
-        for event in annotation.events
-            bytes_written += Base.write(io, event, 0x14)
-        end
-    end
-    bytes_written += Base.write(io, 0x0)
-    return bytes_written
-end
-
-sign(annotation::AbstractAnnotation) = annotation.offset >= 0 ? '+' : '-'
-
-has_duration(annotation::RecordAnnotation) = false
-has_duration(annotation::TimestampAnnotation) = annotation.duration !== nothing
-
-duration(annotation::RecordAnnotation) = nothing
-duration(annotation::TimestampAnnotation) = annotation.duration
-
-footer(annotation::RecordAnnotation) = (0x14, 0x14)
-footer(annotation::TimestampAnnotation) = 0x14
-
-has_events(annotation::RecordAnnotation) = annotation.events !== nothing
-has_events(annotation::TimestampAnnotation) = true
-
-function write_padded(io::IO, value, byte_limit::Integer; pad::UInt8=0x20)
-    bytes_written = write_bytes(io, value)
+function edf_write(io::IO, value, byte_limit::Integer; pad::UInt8=UInt8(' '))
```

Comment:
I think the level of refactoring you've done means that `pad` doesn't need to be a kwarg anymore, given that it was used for edge cases that no longer exist, since `write_signal_record` does this on its own.